File: core/PalindromeNumber/PalindromeNumber.py
def isPalindrome(x: int) -> bool:
    if x < 0 or (x % 10 == 0 and x != 0):
        return False

    reverted_num = 0
    while x > reverted_num:
        reverted_num = reverted_num * 10 + x % 10
        x //= 10

    return x == reverted_num or x == reverted_num // 10

File: core/PalindromeNumber/PalindromeNumber.java
package com.dbc.code;

public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) return false;
        int revertedNum = 0;
        while (revertedNum < x) {
            revertedNum = revertedNum * 10 + x % 10;
            x /= 10;
        }
        return revertedNum == x || x == revertedNum / 10;
    }
}
